packages/components/order-tracking/src/order-tracking.controller.js

```diff
@@ -0,0 +1,143 @@
+import angular from 'angular';
+import flatten from 'lodash/flatten';
+import map from 'lodash/map';
+import reverse from 'lodash/reverse';
+
+const ORDER_DEFAULT_HISTORY_LENGTH = 3;
+const SECONDS = 1000;
+const ORDER_DETAILS_POLLING_INTERVAL = 60 * SECONDS;
+const ORDER_FOLLOW_UP_POLLING_INTERVAL = 60 * SECONDS;
+
+export default class OrderTrackingController {
+  /* @ngInject */
+  constructor($log, $q, $timeout, OvhApiMeOrder) {
+    this.$log = $log;
+    this.$q = $q;
+    this.$timeout = $timeout;
+    this.OvhApiMeOrder = OvhApiMeOrder;
+    this.maxHistoryLength = ORDER_DEFAULT_HISTORY_LENGTH;
+  }
+
+  $onInit() {
+    return this.refresh();
+  }
+
+  refresh() {
+    this.order = null;
+    this.orderDetails = null;
+    this.orderFollowUp = null;
+    this.orderHistory = null;
+    this.polling = {
+      orderDetails: null,
+      orderFollowUp: null,
+    };
+    this.error = null;
+    this.isLoading = true;
+
+    this.OvhApiMeOrder.v6().resetAllCache();
+    this.cancelOrderDetailsPolling();
+    this.cancelOrderFollowUpPolling();
+
+    return this.OvhApiMeOrder.v6().get({
+      orderId: this.orderId,
+    }).$promise.then(order => this.getOrderStatus(order).then(({ status }) => {
+      const orderWithStatus = order;
+      orderWithStatus.status = status;
+      return orderWithStatus;
```

Comment:
Maybe you can have a `Order.class` for this

Reply:
it would make sense in the case of a Service that expose data to return an instance of a class for example ; but imho i the case of a controller i don't think it's better. Since JS objects are dynamics there is no need to create a new class each time you want to alter an object